tests/lib.sh

```diff
@@ -36,12 +39,23 @@ check_signature () {
   rlen=$((${#r_val}/2))
   slen=$((${#s_val}/2))
 
-  rfmt="02$(printf "%x" $rlen)${r_val}"
-  sfmt="02$(printf "%x" $slen)${s_val}"
+  rfmt="02 $(printf "%x" $rlen) ${r_val}"
+  echo $rfmt > RFMT
+  sfmt="02 $(printf "%x" $slen) ${s_val}"
+  echo $sfmt > SFMT
 
-  SIG="30$(printf "%x" $(($rlen+$slen+4)))$rfmt$sfmt"
+  SIG="30 $(printf "%x" $(($rlen+$slen+4))) $rfmt $sfmt"
+  echo $SIG > sigdump
 
-  xxd -r -ps <<<"$1" | blake2b_lock_hash | xxd -p -r | openssl pkeyutl -verify -pubin -inkey tests/public_key_0_0.pem -sigfile <(xxd -r -ps <<<"$SIG")
+  xxd -r -ps <<<"$2" | tee dumpfile0 | blake2b_p | tee hashfile | xxd -p -r | openssl pkeyutl -verify -pubin -inkey <(get_key_in_pem $1) -sigfile <(xxd -r -ps <<<"$SIG")
+}
+
+get_key_in_pem() {
+  derivation=${1:-8002000015058000002c80000135800000000000000000000000}
+  result="$(apdu_with_clicks $derivation "rR" |& egrep "b'41.*'9000" | sed "s/^<= b'41//;s/'9000$//")"
```

Comment:
Nice! Will test this on my ledger

Reply:
> public_key_der_prefix.der looks like it was added by mistake.

public_key_der_prefix.der is the prefix to convert from the raw key provided by the ledger to the ASN.1 format that openssl uses; we definitely need it. Could rename it.